Build failure: spotdl

### Steps To Reproduce

Steps to reproduce the behavior:
1. build spotdl

### Build log

```
=========== 15 failed, 41 passed, 7 deselected, 6 warnings in 4.77s ============
/nix/store/wr08yanv2bjrphhi5aai12hf2qz5kvic-stdenv-linux/setup: line 1559: pop_var_context: head of shell_variables not a function context
error: builder for '/nix/store/sqlncv0041gjzxpjpf129w06cdi4hj6j-spotdl-4.2.1.drv' failed with exit code 1;
       last 10 log lines:
       > FAILED tests/types/test_artist.py::test_artist_from_string - vcr.errors.CannotOverwriteExistingCassetteException: Can't overwrite existi...
       > FAILED tests/types/test_playlist.py::test_playlist_from_url - vcr.errors.CannotOverwriteExistingCassetteException: Can't overwrite existi...
       > FAILED tests/types/test_playlist.py::test_playlist_from_string - vcr.errors.CannotOverwriteExistingCassetteException: Can't overwrite existi...
       > FAILED tests/types/test_playlist.py::test_playlist_length - requests.exceptions.ConnectionError: HTTPSConnectionPool(host='accounts.spo...
       > FAILED tests/types/test_song.py::test_song_from_url - vcr.errors.CannotOverwriteExistingCassetteException: Can't overwrite existi...
       > FAILED tests/utils/test_github.py::test_get_status - requests.exceptions.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
       > FAILED tests/utils/test_github.py::test_check_for_updates - requests.exceptions.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
       > FAILED tests/utils/test_github.py::test_download_github_dir - requests.exceptions.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
       > =========== 15 failed, 41 passed, 7 deselected, 6 warnings in 4.77s ============
       > /nix/store/wr08yanv2bjrphhi5aai12hf2qz5kvic-stdenv-linux/setup: line 1559: pop_var_context: head of shell_variables not a function context
       For full logs, run 'nix-store -l /nix/store/sqlncv0041gjzxpjpf129w06cdi4hj6j-spotdl-4.2.1.drv'.
```

### Additional context

I just tried to install it, doing it from `configuration.nix` or `nix-shell -p` gives the same result. Let me know if I can provide more useful information.

### Notify maintainers

@dotlambda 

### Metadata

```
[andrea@nixos:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.63, NixOS, 24.05 (Uakari), 24.05pre552293.5a09cb4b393d`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

--- a/pkgs/tools/audio/spotdl/default.nix
+++ b/pkgs/tools/audio/spotdl/default.nix
@@ -9,15 +9,14 @@ let
   python = python3;
 in python.pkgs.buildPythonApplication rec {
   pname = "spotdl";
-  version = "4.2.1";
-
-  format = "pyproject";
+  version = "4.2.2";
+  pyproject = true;
 
   src = fetchFromGitHub {
     owner = "spotDL";
     repo = "spotify-downloader";
     rev = "refs/tags/v${version}";
-    hash = "sha256-xKas3WO3uigY1iFfxIN3+d+5U31vM7cLv08oMef8trc=";
+    hash = "sha256-pJr0OGUI3OcFsmvn9eqkvpFeF1EkHDdNoWc91s8h9O8=";
   };
 
   nativeBuildInputs = with python.pkgs; [
@@ -66,6 +65,9 @@ in python.pkgs.buildPythonApplication rec {
     # require networking
     "tests/test_init.py"
     "tests/test_matching.py"
+    "tests/providers/lyrics"
+    "tests/types"
+    "tests/utils/test_github.py"
     "tests/utils/test_m3u.py"
     "tests/utils/test_metadata.py"
     "tests/utils/test_search.py"
@@ -73,7 +75,6 @@ in python.pkgs.buildPythonApplication rec {
 
   disabledTests = [
     # require networking
-    "test_album_from_url"
     "test_convert"
     "test_download_ffmpeg"
     "test_download_song"

